sdk/go/property/map.go

```diff
@@ -22,7 +22,9 @@ import (
 type Map struct{ m map[string]Value }
 
 // AsMap converts the [Map] into a native Go map from strings to [Values].
-func (m Map) AsMap() map[string]Value { return copyMap(m.m) }
+//
+// AsMap always returns a non-nil map.
```

Comment:
Feels a bit inconsistent that empty arrays are nil but empty maps are non-nil. I get this is probably because Go makes it easy to work with nil slices but nil maps can panic but still maybe worth calling out that's the reason for the inconsistency?

Reply:
That was the though. I'll add a comment to that effect.